intro/monetary_values.asciidoc

```diff
@@ -3,30 +3,68 @@
+*To be able to encode or decode monetary values, any client of the API should collect and store the <<Reference_Data, Reference Data>> provided through the <<market_data_protocol, Market Data Service Protocol>> which will indicate the number of decimal places that should be assumed for scaling values in the given deliverable or tradeable entity. This is sent at the beginning of each cycle and updated as needed.*
 
-Here is the example of representation of value "price" equal to "1.12345678 ABC" for some <<deliverable, *deliverable*>> entity:
+As an example consider that we have a tradeable entity described as follows:
 
-The <<price, *price*>> tells you what the value is in.
-The number of decimal places that *price* uses will be determined by
-the <<decimal_places,*decimal_places*>> of the corresponding <<deliverable_id,*price_deliverable_id*>>:
+[source,json]
+{
+    "tradeable_entity": {
+        "tradeable_entity_id": 12345,
+        "symbol": "BTC-USDT 25-Dec-2020 C15000",
+        "price_deliverable_id":2,     // USDT
+        "price_decimal_places":4,
+        "quantity_deliverable_id":10, // Option: BTC-USDT 25-Dec-2020 C15000
+        "quantity_decimal_places":1,
+        // ...
+    }
+}
 
-....
-"price":                112345678,
-"price_deliverable_id": 3,
-....
-where using *price_deliverable_id* a corresponding descriptor of *deliverable* entity
-can be found in the dessiminated reference data:
-....
-"deliverable_id":       3,
-"decimal_places":       8
-..
-....
-so using value of *decimal_places* corresponding integer scaling can be found as 10^8, so resulting encoded value "price" will be "112345678" (= 1.12345678 x 10^8).
+and we receive a message over the <<market_data_protocol, Market Data Service Protocol>> which has an integer  <<price, *price*>> of `2345600` and whose <<tradeable_entity_id,*tradeable_entity_id*>> field corresponds to the tradeable entity shown above, that is, `12345`.
+
+We know from looking up tradeable entity `12345` that the price of the tradeable entity is represented as the corresponding <<deliverable_id,*price_deliverable_id*>> (which we know to be USDT if we look up deliverable_id `2` in our stored deliverables.
+
+We also know that prices listed for this tradeable entity will have `4` decimal places assumed because the value for <<decimal_places,*price_decimal_places*>> is `4`. This means that for the number given of `2345600`, the actual price is `234.5600`.
+
+To convert to and from the scaled integer representation you can use the following simple formulae:
+
+1. From scaled integer to floating point: `2345600 / pow( 10, 4 ) = 234.5600`
+2. From floating point to scaled integer: `234.5600 x pow( 10, 4 ) = 2345600`
+
+The relationship is `scaled_integer` / pow( 10, `decimal_places` ) = `floating_point`.
```

Comment:
I think that in our examples we should not use `floating point` values (and related operations) at all...

Otherwise readers will use as our examples "as-is" and will still have rounding happening on their end, **while our goal is opposite**.

So I suggest to show how to get **whole** and **fractional** parts from scaled integer, using "/" and "%" operators. It can be ported to Python, Java, Go code where `decimal` types exist and suggested for monetary values.

Reply:
What if we expressed the relationship first using mathematical symbols? For example:

2345600 &divide; 10<sup>4</sup>

This means that it cannot be used as-is in any programming language. We can then show specific "safe" code examples using `/` and `%`.

Reply:
addressed in f19c9e186b49260e625928b0749660f1fc779175